2026-05-17T11:27:34.4075142Z ##[group]Run cargo audit
2026-05-17T11:27:34.4075414Z [36;1mcargo audit[0m
2026-05-17T11:27:34.4107997Z shell: /usr/bin/bash -e {0}
2026-05-17T11:27:34.4108229Z env:
2026-05-17T11:27:34.4108405Z   CARGO_HOME: /home/runner/.cargo
2026-05-17T11:27:34.4108630Z   CARGO_INCREMENTAL: 0
2026-05-17T11:27:34.4108824Z   CARGO_TERM_COLOR: always
2026-05-17T11:27:34.4109019Z ##[endgroup]
2026-05-17T11:27:34.4630033Z [1m[92m    Updating[0m crates.io index
2026-05-17T11:27:34.9524272Z [1m[92m     Locking[0m 299 packages to latest Rust 1.95.0 compatible versions
2026-05-17T11:27:35.1607393Z [0m[0m[1m[32m    Fetching[0m advisory database from `https://github.com/RustSec/advisory-db.git`
2026-05-17T11:27:37.8726240Z [0m[0m[1m[31merror:[0m error loading advisory database: parse error: error parsing /home/runner/.cargo/advisory-db/crates/libcrux-ml-dsa/RUSTSEC-2026-0076.md: parse error: TOML parse error at line 5, column 8
2026-05-17T11:27:37.8727392Z   |
2026-05-17T11:27:37.8727645Z 5 | cvss = "CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N"
2026-05-17T11:27:37.8728150Z   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-05-17T11:27:37.8728428Z unsupported CVSS version: 4.0
2026-05-17T11:27:37.8728575Z 
2026-05-17T11:27:37.8753315Z ##[error]Process completed with exit code 1.

--- a/.github/workflows/security.yml
+++ b/.github/workflows/security.yml
@@ -19,7 +19,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout code
-      uses: actions/checkout@v4
+      # Pinned to actions/checkout v6.0.2 (Node.js 24) to avoid the
+      # Node.js 20 deprecation warning that will become a hard error on
+      # GitHub-hosted runners from September 2026 onwards.
+      uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
 
     - name: Install Rust toolchain
       uses: dtolnay/rust-toolchain@stable
@@ -30,7 +33,11 @@ jobs:
       # release of cargo-audit (or any transitive dep) would execute via
       # build.rs with this workflow's issues: write / checks: write
       # GITHUB_TOKEN. See Issue #1223.
-      run: cargo install --locked --version 0.21.2 cargo-audit
+      # Updated to 0.22.1: this version adds CVSS 4.0 support, which is
+      # required to parse new advisory-db entries (e.g. RUSTSEC-2026-0076).
+      # cargo-audit 0.21.x fails with "unsupported CVSS version: 4.0" and
+      # exits 1, breaking the security scan.
+      run: cargo install --locked --version 0.22.1 cargo-audit
 
     - name: Run security audit
       run: cargo audit

